pkg/api/validation/validation.go

```diff
@@ -2696,32 +2696,47 @@ func validateResourceQuotaScopes(resourceQuota *api.ResourceQuota) field.ErrorLi
 func ValidateResourceQuota(resourceQuota *api.ResourceQuota) field.ErrorList {
 	allErrs := ValidateObjectMeta(&resourceQuota.ObjectMeta, true, ValidateResourceQuotaName, field.NewPath("metadata"))
 
-	fldPath := field.NewPath("spec", "hard")
-	for k, v := range resourceQuota.Spec.Hard {
+	allErrs = append(allErrs, ValidateResourceQuotaSpec(resourceQuota.Spec, field.NewPath("spec"))...)
+	allErrs = append(allErrs, ValidateResourceQuotaStatus(resourceQuota.Status, field.NewPath("status"))...)
+
+	return allErrs
+}
+
+func ValidateResourceQuotaStatus(status api.ResourceQuotaStatus, fld *field.Path) field.ErrorList {
```

Comment:
> The same convention was followed for the ValidatePod and ValidatePodSpec.
> I guess why not pointers?

Mutation potential?  Doesn't indicate what the API is actually trying to do?

I don't have any real object, just seems odd and there some that aren't.  I don't mind per-se if you think its more correct.